Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -475,43 +475,58 @@ namespace OpenInfraPlatform {
 				std::vector<carve::geom::vector<2>> outer_loop;
 
 				bool horizontal = profileDef->HorizontalWidths;
-				vector<double> b = profileDef->Widths * UnitConvert()->getLengthInMeterFactor();
-				vector<double> slope = profileDef->Slopes * UnitConvert()->getAngleInRadianFactor();
 
-				if (sizeof(b) != sizeof(slope)) {
-					throw oip::InconsistentModellingException(profileDef, "Number of Widths is not even to number of Slopes");
+				std::vector<double> widths;
+				widths.reserve(profileDef->Widths.size());
+				std::transform(
+					profileDef->Widths.begin(),
+					profileDef->Widths.end(),
+					widths.begin(),
+					[](auto& it) {return it * UnitConvert()->getLengthInMeterFactor(); });
+
+				std::vector<double> slopes;
+				slopes.reserve(profileDef->Slopes.size());
+				std::transform(
+					profileDef->Slopes.begin(),
+					profileDef->Slopes.end(),
+					slopes.begin(),
+					[](auto& it) {return it * UnitConvert()->getAngleInRadianFactor(); });
+
+				if (widths.size() != slopes.size()) {
+					throw oip::InconsistentModellingException(profileDef, "Number of Widths is not equal to the number of Slopes");
 				}
 				else {
 					double TagX = 0.0;
 					double TAgY = 0.0;
 
-					for (int i = 0; i < sizeof(b) / 8; i++) {
+					for (int i = 0; i < widths.size(); i++) {
 						paths.push_back(carve::geom::VECTOR(TagX, TagY));
-						double x, y = AddXYCoordinates(horizontal, b(i), slope(i));
+						std::tuple<double,double>(x, y) = AddXYCoordinates(horizontal, widths[i], slopes[i]);
 						TagX = TagX + x;
 						TagY = TagY + y;
 					}
 
-				paths.push_back(carve::geom::VECTOR(TagX, TagY));
+					paths.push_back(carve::geom::VECTOR(TagX, TagY));
 				}
 			}
 
-			double AddXYCoordinates(bool horizontal, double b, double slope) const {
+			std::tuple<double,double> AddXYCoordinates(const bool& horizontal, const double& width, const double& slope) const throw(...) {

```

Comment:
Proposal on function name: It calculates xy coordinates, not adds them. So call it `CalculateXYfromPolar`

Reply:
Resolved: c713fd2